### PR TITLE
Use wrapGAppsHook where possible

### PR DESCRIPTION
**Pull request by fpletz**
This PR adds wrapping `GST_PLUGIN_SYSTEM_PATH` to `wrapGAppsHook` and cleans up a lot of packages to use `wrapGAppsHook` instead of creating the wrapper and specifying the environment variables manually.

This fixes missing icons in some applications. One application even failed to run previously.

Currently, this is still a work in progress because I haven't tested all applications yet.

### CODE REVIEW

**Comment by jgeerds** [comment]
:+1: good idea!

**Comment by vcunat** [comment]
Patching gnome 3.16 stuff, note that these are no longer the default and the improvements will die soon. The upgrading workflow is to copy the latest files (3.18 ATM) and edit them, I believe.

File: pkgs/applications/video/pitivi/default.nix [hunk 1]
```diff
@@ -40,11 +40,11 @@ in stdenv.mkDerivation rec {
...
   ]);
 
-  preFixup = ''
-    wrapProgram "$out/bin/pitivi" \
-      --set GDK_PIXBUF_MODULE_FILE "$GDK_PIXBUF_MODULE_FILE" \
-      --prefix GI_TYPELIB_PATH : "$GI_TYPELIB_PATH" \
-      --prefix GST_PLUGIN_SYSTEM_PATH_1_0 : "$GST_PLUGIN_SYSTEM_PATH_1_0" \
-      --prefix XDG_DATA_DIRS : "$XDG_ICON_DIRS:$out/share:$GSETTINGS_SCHEMAS_PATH"
-  '';
+#  preFixup = ''
```
Comment (lethalman) [inline pkgs/applications/video/pitivi/default.nix]: Why commented?

**Comment by lethalman** [comment]
Great work thanks! Note however that GNOME 3.16 has been dropped.

File: pkgs/applications/video/pitivi/default.nix (hunk 1 above)
Comment (fpletz) [inline pkgs/applications/video/pitivi/default.nix]: This just means that this package builds but still has issues or hasn't been tested yet, so this just for reminding me. When I'm finished all commented code will be removed.

**Comment by zimbatm** [comment]
ping

**Comment by fpletz** [comment]
Abandoned. Some packages were already fixed.

**Pull request closed without merging**